lib/src/common/disposable_manager.dart

```diff
@@ -186,3 +257,30 @@ abstract class LeakFlagger {
 /// the object managing it is disposed, the future will complete with
 /// an instance of this exception.
 class ObjectDisposedException implements Exception {}
+
+/// Used to invoke, and remove references to, a [Disposer] before disposal of the
+/// parent object.
+abstract class SimpleDisposable {
```

Comment:
@todbachman-wf I'm not really following. The doc comment seems to be exactly what it's used for  from a consumer perspective. We use it internally as well but I did that just to get rid of the `_Disposable` class. Wondering how you thought this might be expanded?